tests/neg-custom-args/fatal-warnings/structural.scala

```diff
@@ -67,7 +67,7 @@ object Test {
 package p6 {
 
   class Refinements {
-    val y: { val x: T; type T }  // error: deprecated warning: illegal forward reference in refinement; now illegal
+    val y: { val x: T; type T }  // illegal forward reference in refinement; now illegal
```

Comment:
We should move this test to `tests/neg-custom-args/deprecation`

Reply:
Upon reflection, I think maybe we should always emit this warning, as it's a language feature warning.